### PR TITLE
[FEAT] Fishing Updates

### PR DESCRIPTION
**Pull request by adamhannigan**
# Description

This PR introduces the first set of fishing updates we have planned. Including:

- Beach Fishing
- Beach/Reef Fish (Halibut, ParrotFish, AngelFish)
- Deep Sea Helm Milestone - Catch 5 of every fish > Increases Marine Marvel Chance

Screenshot 2023-12-18 at 5 27 16 pm
Screenshot 2023-12-18 at 5 32 18 pm

### CODE REVIEW

File: src/features/game/types/milestones.ts [hunk 1]
```diff
@@ -133,6 +134,24 @@ export const FISH_MILESTONES: Record<MilestoneName, Milestone> = {
...
   },
+  "Deep Sea Diver": {
+    task: "Catch 5 of every fish",
+    percentageComplete: (farmActivity: GameState["farmActivity"]) => {
+      const encyclopediaFish = getEncyclopediaFish();
+      const totalFishRequired = encyclopediaFish.length * 5;
+
+      const totalFishCaught = encyclopediaFish.reduce(
+        (total, name) =>
+          total + Math.min(farmActivity[`${name} Caught`] ?? 0, 1),
```
Comment (spencerdezartsmith) [inline src/features/game/types/milestones.ts:145]: The logic here is incorrect. We will always be adding `1` to the total if we take the `min` value. 
```suggestion
          total + (farmActivity[`${name} Caught`] ?? 0)),
```

**Review by spencerdezartsmith — APPROVED** [review]
🎣 Nice one! All is working well for me. Love the pub-sub pattern.

There was some incorrect logic for the new milestone. Left a suggestion. I will approve for now. Just need to fix that up.

File: src/features/game/types/milestones.ts (hunk 1 above)
Comment (brian-man) [reply src/features/game/types/milestones.ts:145]: @spencerdezartsmith That proposed change will mean that you just need to catch `encyclopediaFish.length * 5` anchovies to complete the milestone.

I think the logic he was looking for is this...
```
total + Math.min(farmActivity[`${name} Caught`] ?? 0, 5),
```
... which limits the contribution for any fish species to 5.

**Comment by Dionis404** [comment]
The Deep Sea Helm bonus is not showing up in the game, does it really increase the chance of catching Marine Marvel ?

**Pull request merged**